src/core/orderbookutils.js

```diff
@@ -23,7 +40,16 @@ function assertValidNumber(number, message) {
 OrderBookUtils.getOfferTakerGetsFunded = function(offer) {
   assertValidNumber(offer.taker_gets_funded, 'Taker gets funded is invalid');
 
-  return Amount.from_json(offer.taker_gets_funded + IOU_SUFFIX);
+  let currency = offer.TakerPays.currency;
```

Comment:
This seems a little redundant at first glance and repeats the solution in many places. Previously the code was using `IOU_SUFFIX` to create a non native amount from what's essentially just the value.

AFAIK it doesn't matter what currency/issuer the amount has, just that it has a quantity and can interoperate with the Amount class math as an IOU. 

I think the goal here was to remove the need for having to tack on a string to force a value to parse as an IOU, and not native. `Amount.from_number` just happens to work for this case I think, as it sets the currency/issuer to 1/1 respectively, which are neutral.  

What `number` actually means in `from_number` is unclear. Currently, it just happens to work with anything that is valid for the BigNumber constructor. The value here (that previously had IOU_SUFFIX tacked on) would be a legitimate numeral, but as a string not a `Number`. 

Depending on how strong your need for order, you could add an `Amount.from_neutral` or the like :) 

`Amount.from_json(offer.taker_gets_funded + IOU_SUFFIX);` 

->

`Amount.from_number(offer.taker_gets_funded);`

@wltsmrz  ?